scipy/optimize/linprog_ip.py

```diff
@@ -0,0 +1,1994 @@
+# -*- coding: utf-8 -*-
+"""
+Created on Thu Feb 02 17:09:17 2017
+
+@author: Matt
+
+To do:
+multiple presolve
+questions for scipy-dev
+* This is written in Python 2. How should I check compatibility with Python 3?
+* Are the warnings OK? Do I need an option to disable them?
+* Is 100% pep8 compliance necessary?
+* How should unit tests be performed on both simplex and ip?
+* Can I leave multi-pass presolve to an update?
+* Can I leave sparse presolve to an update?
+* Do I need to implement a callback interface?
+* There are some changes to be made to simplex; it should use the ip presolve
+ routine. Can that be left to an update?
+* How do I document options not common with simplex?
```

Comment:
You can check if the documentation changes worked OK by going to
https://circle-artifacts.com/gh/scipy/scipy/2294/artifacts/0/home/ubuntu/scipy/doc/build/html-scipyorg/index.html

The number "2294" should be replaced by the number in the link you see in the "ci/circleci" details link at the bottom of this page on github. (Not super user-friendly, though...)